### ISSUE TITLE
Suppress warning with valid 

### ISSUE THREAD

**Issue by dholliday3** [wagtail/wagtail#8762]
![Screen Shot 2022-06-23 at 6 20 37 PM](https://user-images.githubusercontent.com/15958912/176035673-2b8b43ce-c7c0-41a6-bdac-2e3924092d9b.png)

I'm getting this warning for a valid many-to-many field that is being added as a filter to `search_fields.` 
The field is referenced from `ParentalManyToManyField`. I tried using `RelatedField` as proposed in the [docs](https://docs.wagtail.org/en/stable/topics/search/indexing.html#index-relatedfields) which doesn't work. 

Is this a known error? What could be the workaround? 

In my case this warning is clogging up my logs in production but without a log `id` I'm unable to suppress the warnings in my Django app. 

Is it possible to add a id to the warning so we can suppress this warning on valid many-to-many relationships? 
- [Example warning logs](https://www.programcreek.com/python/example/97878/django.core.checks.Warning)
- [Django Docs for warning/error Ids](https://docs.djangoproject.com/en/3.1/ref/checks/#model-fields)

### Steps to Reproduce

```
class TaxonomyCategory(models.Model):
    name = models.CharField(max_length=100, blank=False)
    slug = models.CharField(max_length=100, blank=False, unique=True)
    description = models.CharField(max_length=255, blank=False)
    group = models.CharField(max_length=100, blank=False, default="")
    
class Page(RoutablePageMixin, BrandMixin, Page):
    taxonomy_categories = ParentalManyToManyField(
        CodeExchangeTaxonomyCategory, blank=True
    )

    search_fields = Page.search_fields + [
        index.FilterField('taxonomy_categories'),
        index.FilterField('group'),
    ]
   
```

### Technical details

-   Python version: 3.8
-   Django version: 3.1.13
-   Wagtail version: 2.13.5
-   Postegres version: 11
-   Browser version: N/A

**Comment by gasman** [comment]
Hi @dholliday3 - thanks for the report.

Have added an ID to the warning in #8780 - will leave the other issue (i.e. whether m2m relations should be considered valid for FilterField) to someone more knowledgeable about the search module.

**Comment by javoweb** [comment]
Hey @gasman. QQ. Will this patch be available in the next LTS minor version? 2.15.x?

**Comment by gasman** [comment]
@javoweb No, LTS releases will generally only receive updates for stability and security fixes, and this is really too minor for that.

**Comment by gasman** [comment]
> will leave the other issue (i.e. whether m2m relations should be considered valid for FilterField) to someone more knowledgeable about the search module.

["That was a problem for future me, and now I am future me."](https://www.youtube.com/watch?v=nzDqvxiEZ4k)

This is somewhat related to #6616, although I'll consider it separate as it doesn't involve traversing a `RelatedFields` definition - it's possible to construct a query like

    Page.objects.filter(taxonomy_categories=some_category)

so it would be reasonable to expect that defining `FilterField("taxonomy_categories")` would allow extending that to

    Page.objects.filter(taxonomy_categories=some_category).search("foo")

Moving this to the wagtailsearch repo.
